languageserver/src/main/scala/langserver/utils/ScodecUtils.scala

```diff
@@ -0,0 +1,45 @@
+package langserver.utils
+
+import java.nio.charset.StandardCharsets.US_ASCII
+import scodec._
+import scodec.bits._
+import scodec.codecs._
+
+object ScodecUtils {
+  //parse out ascii string ignoring prefix and terminator
+  def bracketedBy(prefix: String, terminator: String): Codec[String] = filtered(ascii, new Codec[BitVector] {
```

Comment:
Scodec did not have a nice way to read/write a value between two string, this helps keep the remainder of the codec simple.  I may be able to simplify this with another pass.